src/main/java/bg/libapp/libraryapp/exceptions/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package bg.libapp.libraryapp.exceptions;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler
+        extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value
+            = {IllegalArgumentException.class, IllegalStateException.class, BookNotFoundException.class, UserWithThisUsernameAlreadyExistsException.class, CannotProcessJsonOfEntity.class, AuthorNotFoundException.class, FieldNotFoundException.class, GenreNotFoundException.class, BookAlreadyAddedException.class, UserNotFoundException.class, UserIsAlreadyActivatedException.class, UserIsAlreadyDeactivatedException.class})
+    protected ResponseEntity<Object> handleConflict(
```

Comment:
You should have different methods for the different types of exceptions - ex. handleNotFound, handleBadRequest etc.